apps/prairielearn/src/pages/instructorCourseAdminStaff/instructorCourseAdminStaff.js

```diff
@@ -269,12 +287,15 @@ ${given_cp_and_cip.join(',\n')}
       // in the given course instance. We choose not to do this for the same
       // reason as above (see handler for course_permissions_update_role).
 
+      const course_instances = await selectAuthorizedCourseInstancesForCourse({
+        course_id: res.locals.course.id,
+        user_id: res.locals.user.user_id,
+        authn_user_id: res.locals.authn_user.user_id,
+        is_administrator: res.locals.is_administrator,
+      });
+
       if (req.body.course_instance_id) {
-        if (
-          !res.locals.authz_data.course_instances.find((ci) =>
-            idsEqual(ci.id, req.body.course_instance_id),
-          )
-        ) {
+        if (!course_instances.find((ci) => idsEqual(ci.id, req.body.course_instance_id))) {
```

Comment:
Is it worth searching all course instances and then filtering on the current instance? Would it be more efficient to have a separate query that checks if the specific instance is authorized? Same with other check below in this file. I would be fine if this check is in a helper function that for now only does the same, with a future PR handling the actual efficient search (possibly after the sproc is converted), but this looks wasteful.